### ISSUE TITLE
Deprecations from `tabtab@1.3.2`

### ISSUE THREAD

**Issue by MikeMcC399** [yeoman/yo#815]
## Type of issue

Bug

## My environment

* *OS version/details*: Ubuntu `24.04.1` LTS
* *Node version:* ~`v20.18.0`~ `v22.13.1`
* *npm version:* ~`10.8.2`~ `10.9.2`
* *Version of yo :* ~`5.0.0`~ `5.1.0`

## Expected behavior

When executing

```shell
npm install yo -g
```

there should be no deprecation warnings output.

## Current behavior

When executing

```shell
npm install yo -g
```

the following warnings are output

```text
npm warn deprecated npmlog@2.0.4: This package is no longer supported.
npm warn deprecated are-we-there-yet@1.1.7: This package is no longer supported.
npm warn deprecated gauge@1.2.7: This package is no longer supported.
npm warn deprecated boolean@3.2.0: Package no longer supported. Contact Support at https://www.npmjs.com/support for more info.
```

Three of the above deprecations warnings are caused by [tabtab@1.3.2](https://www.npmjs.com/package/tabtab/v/1.3.2) released 8 years ago.

[tabtab@1.3.2](https://www.npmjs.com/package/tabtab/v/1.3.2) depends on the deprecated [npmlog](https://www.npmjs.com/package/npmlog).

## Steps to reproduce the behavior

Execute

```shell
npm install yo -g
```

## Command line output

```text
npm warn deprecated npmlog@2.0.4: This package is no longer supported.
npm warn deprecated are-we-there-yet@1.1.7: This package is no longer supported.
npm warn deprecated gauge@1.2.7: This package is no longer supported.
npm warn deprecated boolean@3.2.0: Package no longer supported. Contact Support at https://www.npmjs.com/support for more inf

$ npm ls gauge -g
└─┬ yo@5.1.0
  └─┬ tabtab@1.3.2
    └─┬ npmlog@2.0.4
      └── gauge@1.2.7

$ npm ls are-we-there-yet -g
└─┬ yo@5.1.0
  └─┬ tabtab@1.3.2
    └─┬ npmlog@2.0.4
      └── are-we-there-yet@1.1.7
```

## Suggestion

Update to [tabtab@3.0.2](https://www.npmjs.com/package/tabtab/v/3.0.2) which no longer depends on [npmlog](https://www.npmjs.com/package/npmlog).

## Related

- https://github.com/yeoman/yo/pull/595 which was an abandoned attempt to update `tabtab`
- https://github.com/yeoman/yo/issues/508
- Dependabot https://github.com/yeoman/yo/pull/703 has also been trying since 2021

**Comment by MikeMcC399** [comment]
Updated for [yo@5.1.0](https://github.com/yeoman/yo/releases/tag/v5.1.0)

**Comment by JoshuaKGoldberg** [comment]
Indeed. Duplicate of #508. Definitely accepting PRs. Whether it's easier to update to 2 first, then 3, or just skip straight to 3 is TBD. Thanks for reporting! 🎩
